### PR TITLE
feat: all public SDK functions using background queue

### PR DESCRIPTION
**Pull request by levibostian**
PR that partially closes https://github.com/customerio/issues/issues/6675. There will be multiple PRs to get that milestone crossed. I am making multiple PRs to make the (large) milestone easier to review. 

Until https://github.com/customerio/customerio-android/pull/68 is merged, [this](https://github.com/customerio/customerio-android/compare/levi/levi/background-q-hooks...levi/levi/background-q-implement-into-sdk) will give you the changes only in this PR. 

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-android/develop/docs/dev-notes/GIT-WORKFLOW.md). 
- [x] merge https://github.com/customerio/customerio-android/pull/68
- [x] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request. 
- [x] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1). 
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request and it will automatically be merged.

### CODE REVIEW

File: messagingpush/src/main/java/io/customer/messagingpush/MessagingPush.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import io.customer.messagingpush.hooks.MessagingPushModuleHookProvider
+import io.customer.sdk.CustomerIO
+import io.customer.sdk.CustomerIOInstance
+import io.customer.sdk.di.CustomerIOComponent
+import io.customer.sdk.hooks.HookModule
+import io.customer.sdk.hooks.hooks.ProfileIdentifiedHook
+import io.customer.sdk.repository.PreferenceRepository
+import io.customer.sdk.util.Logger
+
+interface MessagingPushInstance {
```
Comment (Shahroz16) [inline messagingpush/src/main/java/io/customer/messagingpush/MessagingPush.kt:15]: I am not sure about this, so we are saying if a user just wants to add a device to its identified customer and do nothing else, he will still need to add the `messaging` module? 

If they have push notification handling on their own, and just want to track metrics, they won't be able to do that either, without adding the module. 

These basic functionality should just be present in sdk module ?

Comment (Shahroz16) [inline messagingpush/src/main/java/io/customer/messagingpush/MessagingPush.kt:15]: I am not sure about this, so we are saying if a user just wants to add a device to its identified customer and do nothing else, he will still need to add the `messaging` module? 

If they have push notification handling on their own, and just want to track metrics, they won't be able to do that either, without adding the module. 

These basic functionality should just be present in sdk module ?

**Comment by levibostian** [comment]
PR blocked until https://github.com/customerio/customerio-android/pull/68 is unblocked

**Comment by levibostian** [comment]
No longer blocked and ready for review.

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**